### PR TITLE
Issue 5157: Add CO2 saving field for Electricity in TH

### PR DESCRIPTION
**Pull request by rhernandez-intertech**
connects #5157

### CODE REVIEW

File: src/app/treasure-hunt/operation-costs/operation-costs.component.html [hunk 1]
```diff
@@ -269,4 +281,34 @@ <h3>Annual Utility Usage and Costs</h3>
...
+        <button class="close" aria-label="Close" (click)="hideZipCodeModal()">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div #modalBody class="modal-body p-5">
+        <app-assessment-co2-savings
+          [co2SavingsData]="co2SavingsData"
+          [isFormDisabled]="false"
+          [inBaseline]="true"
+          (emitUpdateCo2SavingsData)="updatePsatCo2SavingsData($event)">
```
Comment (nbintertech) [inline src/app/treasure-hunt/operation-costs/operation-costs.component.html:300]: Shouldn't say 'psat'

File: src/app/treasure-hunt/operation-costs/operation-costs.component.ts [hunk 2]
```diff
@@ -37,12 +53,28 @@ export class OperationCostsComponent implements OnInit {
       this.saveSettings();
     }
     this.treasureHuntSub.unsubscribe();
+    this.zipCodeModalSub.unsubscribe();
+  }
+
+  ngAfterViewInit() {
+    this.zipCodeModalSub = this.zipCodeModal.onShown.subscribe(() => {
```
Comment (nbintertech) [inline src/app/treasure-hunt/operation-costs/operation-costs.component.ts:60]: This should be in or called from in ngOnInit()

**Review by nbintertech — CHANGES_REQUESTED** [review]
Structure and implementation in general look good. Other than my comments, we need a few more changes:
- Try the modal at one width size smaller
- I don't think we need three action buttons - should just be 'save' or 'apply' 
- Let's get a header on that modal to give some context for the user - ask Kristina. Could just say 'Egrid zipcode lookup' or whatever we're using in the header of psat/phast, 'CO2 Emissions' 
- Don't need a default for zipcode on a new TH
- Default zipcode isn't coming from global settings

File: src/app/treasure-hunt/operation-costs/operation-costs.component.ts (hunk 2 above)
Comment (rhernandez-intertech) [reply src/app/treasure-hunt/operation-costs/operation-costs.component.ts:60]: Everywhere else in the app when a modal subscription is called its called in ngAfterViewInit()

**Review by nbintertech — APPROVED** [review]

**Pull request merged**